pages/expenses.tsx

```diff
@@ -0,0 +1,319 @@
+  query: omitBy(
+    {
+      offset: parseInt(query.offset) || undefined,
+      limit: parseInt(query.limit) || undefined,
+      type: has(expenseTypes, query.type) ? query.type : undefined,
+      status: has(ExpenseStatus, query.status) || query.status === 'READY_TO_PAY' ? query.status : undefined,
+      payout: has(PayoutMethodType, query.payout) ? query.payout : undefined,
+      direction: query.direction,
+      period: query.period,
+      amount: query.amount,
+      tag: query.tag,
+      searchTerm: query.searchTerm,
+      orderBy: query.orderBy,
+    },
+    isNil,
+  ),
+});
+
+const getVariablesFromQuery = query => {
+  const props = getPropsFromQuery(query);
+  const amountRange = parseAmountRange(props.query.amount);
+  const { from: dateFrom, to: dateTo } = parseDateInterval(props.query.period);
+  const showSubmitted = props.query.direction === 'SUBMITTED';
+  const fromAccount = showSubmitted ? { slug: props.collectiveSlug } : null;
+  const account = !showSubmitted ? { slug: props.collectiveSlug } : null;
+  return {
+    collectiveSlug: props.collectiveSlug,
+    fromAccount,
+    account,
+    offset: props.query.offset || 0,
+    limit: props.query.limit || EXPENSES_PER_PAGE,
+    type: props.query.type,
+    status: props.query.status,
+    tags: props.query.tag ? (props.query.tag === 'untagged' ? null : [props.query.tag]) : undefined,
+    minAmount: amountRange[0] && amountRange[0] * 100,
+    maxAmount: amountRange[1] && amountRange[1] * 100,
+    payoutMethodType: props.query.payout,
+    dateFrom,
+    dateTo,
+    orderBy: props.query.orderBy && parseChronologicalOrderInput(props.query.orderBy),
+    searchTerm: props.query.searchTerm,
+  };
+};
+
+type ExpensesPageProps = {
+  collectiveSlug: string;
+  parentCollectiveSlug: string;
+  data: Partial<ExpensesPageQuery>;
+  error?: any;
+};
+
+export const getServerSideProps: GetServerSideProps<ExpensesPageProps> = async ctx => {
+  const props = getPropsFromQuery(ctx.query);
+  const variables = getVariablesFromQuery(ctx.query);
+
+  // Fetch data from GraphQL API for SSR
+  const client = initClient();
+  const { data, error } = await client.query({
+    query: expensesPageQuery,
+    variables,
+    context: API_V2_CONTEXT,
+    fetchPolicy: 'network-only',
+    errorPolicy: 'ignore',
+  });
+
+  return {
+    props: {
+      ...props,
+      data,
+      error: error || null,
+    },
+  };
+};
+
+export default function ExpensesPage(props: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const intl = useIntl();
+  const router = useRouter();
+  const { LoggedInUser } = useLoggedInUser();
+
+  const [fetchData, query] = useLazyQuery(expensesPageQuery, {
+    variables: getVariablesFromQuery(router.query),
+    context: API_V2_CONTEXT,
+  });
+
+  useEffect(() => {
+    if (LoggedInUser) {
+      fetchData();
+    }
+  }, [LoggedInUser]);
+
+  const error = query?.error || props.error;
+  const data: ExpensesPageQuery = query?.data || props.data;
```

Comment:
Any chance we could pass the `client` from `getServerSideProps` and just call `refetch`? That would:
- Make the code shorter
- Prevent inconsistencies (e.g. different variables or context)
- Reduce memory usage, even though I hope Apollo's reusing the same cached objects in this situation

Reply:
Not really. The result of `getServerSideProps` is serialized and passed/loaded as JSON to the frontend. On top of that, there are cases where the variables used in the frontend are going to be different from the ones in the backend (you receive the SSR of the default list but you decide to filter for something else.

Outside the code complexity aspect, getServerSideProps only runs on the backend and the other query only runs on the frontend, which means I don't think they'll share cached objects.